source-postgres/replication.go

```diff
@@ -187,7 +187,7 @@ const standbyStatusInterval = 10 * time.Second
 // The buffer is much larger for PostgreSQL than for most other databases, because
 // empirically this helps us to cope with spikes of intense load which can otherwise
 // cause the database to cut us off.
-var replicationBufferSize = 64 * 1024 // Assuming change events average ~2kB then 64k * 2kB = 128MB
+var replicationBufferSize = 16 * 1024 // Assuming change events average ~2kB then 16k * 2kB = 32MB
```

Comment:
The comment above refers to coping with load spikes which could otherwise cause the database to cut us off - I don't have any kind of feel for how big the buffer needs to be to prevent this from happening, so I can only note that making the buffer smaller feels like it would move closer to that point, whatever it may be.

Is the replication buffer used in backfills? Since we're seeing memory issues with backfills, I'm wondering if the replication buffer would have any expectation of impacting backfills.

Reply:
The load spikes I'm referring to were very hard to hit back when I was doing throughput optimization in the past, and basically require a dedicated database stress-test setup to reproduce.

The replication buffer is always in use, although its fullness will vary depending on how busy the source database is. In theory it fills up while we're performing backfill queries (because the core backfilling-and-merging behavior is necessarily synchronous) and then gets drained once we start waiting for the next watermark change.